### ISSUE TITLE
Solve PPC WebKit Javascript Engine Endianess Problems inherited from Webkit X86!

### ISSUE THREAD

**Issue by raf1024kb** [deadwood-pl/OdysseyWebBrowser#1]
Title says all!

**Comment by AbelSoul** [comment]
Hi folks,
Really sorry I haven't been able to contribute so far. I've been tied up with other stuff over the last few weeks and only tried to set up my A1 again over the weekend. Unfortunately it wouldn't boot but I'm hoping I just need a new CMOS battery, which I'll try to pick up this week.
Cheers,Robert 

    On Monday, 8 February 2016, 12:32, raf1024kb <notifications@github.com> wrote:
 

 Title says all!—

**Comment by kas1e** [comment]
> Title says all!

I feel myself dumb , but what you trying to say ?:) Did you solve those welknown endianes issues ? Or someone else already solve that welknown issues in webkit repo and you want our attention to merge them ?:)

**Comment by raf1024kb** [comment]
Bountysource needed official issue report on Github to create a new bounty,
so I joined OWB repository and created one.
If you want it assigned to you or to a designed team just tell me and I
will assign it to you.
Else you can edit Issue and  assign it to yourself.

Ciao,

Raf
Il 08/feb/2016 14:32, "kas1e" <notifications@github.com> ha scritto:

**Comment by kas1e** [comment]
> Bountysource needed official issue report on Github to create a new bounty,
> so I joined OWB repository and created one.
> If you want it assigned to you or to a designed team just tell me and I
> will assign it to you.
> Else you can edit Issue and assign it to yourself.

I was just in hope it is something already done , not just about making a bounty :)

**Comment by raf1024kb** [comment]
Well... Anyone should contribute with his own abilities in keeping alive
AmigaOS-like systems and a valuable browser to keep alive our dream.

Deadwood asked if someone was so interested enough in OWB to create a
Bounty in this thread:

http://amigaworld.net/modules/news/article.php?storyid=7623

Well... As long none created any as of today, then I just thought  I could
open it by myself!

So I committed it!
That's it.

Was I wrong, Kas1e?

I was mailing chatting with Deadwood in past days and he wanted me to use
Power2People site...
But I prefer use BountySource that is worldwide visible and could attract
also money from big firms like IBM who want to keep alive PPC platform.

By the way Kas1e, did you spot Bountysource IBM bounty for creating LUA
PPC64 version? They offer 5000 US$.

Ciao,

RAF
Il giorno 08/feb/2016 16:36, "kas1e" <notifications@github.com> ha scritto:

**Comment by raf1024kb** [comment]
I updated start comment to better explain nature of Bounty!

**Comment by kas1e** [comment]
> Was I wrong, Kas1e?

Probably yes. 

Beause for first we should find person who will do so.  Without, money will be only collecting dust, and every donor will be angree as nothing will happens.

There is alos no limit is set, so that mean no one will know how much need, and for who. So, my imho it was too fast to create such a bounty.


> I was mailing chatting with Deadwood in past days and he wanted me to use
> Power2People site...
> But I prefer use BountySource that is worldwide visible and could attract
> also money from big firms like IBM who want to keep alive PPC platform.

Probably he was right, power2people was better choice, as it 100% prove to be ok and clear enough in money trasnfer. How all that crap works with BountySource no one know from us know, and i can't bet on it. So, from my imho, choice of bounty site was also wrong.

You probably better before find someone, then discuss it with everyone, and only after that, when person is find, and know what to do, and know how much he want, create a bounty.

Imho, you was too fast. That money will colelct dust :)

**Comment by raf1024kb** [comment]
I wonder if you Amigan people prefer blabbering and continue crying over
split milk, or rather you want to do some truly concrete to solve our
situations.

Well, I did something concrete by putting money in the face of developers.

Now feel free to blame me for this guilty sin!

Developers in Amigaland are few and in the community we all know who are
the best ones, so it will not difficult to assign bounty task to a good
programmer.

Else we will continue facing a increasing divide with our browser
javascript engine and the rest of the world.

-----

Perhaps speaking of another topic of great interest, I know that in DirOpus
5.92 for MorphOS are still present three main bugs and none wants to solve
it, as MorphOS team rather prefer to improve Ambient GUI Windows Manager to
became similar to Directory Opus, than solving these three trivial bugs.

But at this very moment Directory Opus is a really killer application with
unsurpassed features on any existing computing platform, and we need it
running on MorphOS at its best performance!

I can't ask it to you, Kas1e to commint this MOS Diropus 5.92 bug hunting,
as I know of you as AmigaOS 4 developer and not MorphOS one.

Then sooner or later someone will be forced to start a bounty for that bugs
too...

I am ready for that!

Then we will see if there will be any volunteers to finally solve these
MorphOS Diropus problems when developers could facing proper credits in
their faces.

Do you know if Bountysouce is compatible with Sourceforge issue reports
too?

If answer is positive I could start immediately a new bounty!

Ciao,

Raf
Il 09/feb/2016 09:01, "kas1e" <notifications@github.com> ha scritto:

**Comment by kas1e** [comment]
> I wonder if you Amigan people prefer blabbering and continue crying over
> split milk, or rather you want to do some truly concrete to solve our
> situations.

Creating a bounty on your own, is not what make situation to be solved. You should firstly find a dedicated developer, before making any bounties. Ppls will think that you already "deal" with someone, and send money. And that will be fail, because you found no one, only collect money.
Its wrong way.

Deadwood say you - power2people is better, and he is one who workign on odyssey now. That mean, that what he say, should be done, not in other way. 

> Well, I did something concrete by putting money in the face of developers.

No. You making a bounty, which sounds like you have develioper who will fix all the issues. Bounty done wrong and desribe nothing. We all in amigaland know that when developer is agree, there no problems to make a bounty. 

You was too fast. 

> Now feel free to blame me for this guilty sin!

You was too fast. That bounty will fail as no devloper assinged, and you choice wrong bounty site, as power2people is one we in amigaland use, and which is 100 times checked and proved to be good enough for us.

> Developers in Amigaland are few and in the community we all know who are
> the best ones, so it will not difficult to assign bounty task to a good
> programmer.

You are too fast. Thinkgs not done like this. You first need to find developer, make an agreement with him, about what should be done, how much he want, etc, and only after that create a bounty. Not the other way. 

I know what i say, i deal with it few times :)

> Else we will continue facing a increasing divide with our browser
> javascript engine and the rest of the world.

Find a developer for, and then bounty.

> Then sooner or later someone will be forced to start a bounty for that bugs
> too...
> I am ready for that!

You can be sure, that we can deal with things in fine maner as well. And making a bounty is not problems for any of us. Just if there none, then it mean no developers willing to work on, or have another ideas, or have another timeline. Just not be too fast with things.

> Then we will see if there will be any volunteers to finally solve these
> MorphOS Diropus problems when developers could facing proper credits in
> their faces.

I works already with ppls who can deal with, and there no needs for help like creating an uncertain bounties, sorry :)

> Do you know if Bountysouce is compatible with Sourceforge issue reports
> too?

I know that power2people is best for amiga projects, and not bountysource :)

sorry, i know you want to help , but (and that only my imho), you was too fast. You first should discuss it everywhere, without taking an action by taking money from users

Now, you will be in situation when ppls will ask _you_ about where their money :)

> If answer is positive I could start immediately a new bounty!

No please no ! we can deal with it ourselfs :))

**Comment by raf1024kb** [comment]
AaI just edited again bounty mission info, and added some technical specs on the work to do

**Comment by internetzel** [comment]
Here's my current patch from Leopard WebKit for JavaScriptCore.
It's been done on the branch "safari-601-branch" which was originally branched off at trunk revision 186671.

Built with the attached patch (with full optimization on gcc 5.3) on a 32 bit big endian PowerPC machine the JavaScriptCore C-Loop Interpreter passes all tests that I suppose it should pass.

Most chunks in the patch shouldn't be needed for MorphOS but someone who knows how to deal with it, like Fabien Coeurjoly, will easily be able to strip the unneeded parts out.

[JavaScriptCore_601.zip](https://github.com/deadwood-pl/OdysseyWebBrowser/files/153546/JavaScriptCore_601.zip)

**Comment by raf1024kb** [comment]
Thank you Internetzel!

I will pass this info to developers, so they will verify if it is working
properly and then declare you eligible to get bounty.

Ciao,

Raf
Il 01/mar/2016 21:44, "internetzel" <notifications@github.com> ha scritto:
